Sources/Commands/PackageCommands/APIDiff.swift

```diff
@@ -173,39 +194,180 @@ struct APIDiff: AsyncSwiftCommand {
         }
     }
 
-    private func determineModulesToDiff(packageGraph: ModulesGraph, observabilityScope: ObservabilityScope) throws -> Set<String> {
+    private func runWithIntegratedAPIDigesterSupport(_ swiftCommandState: SwiftCommandState, baselineRevision: Revision, baselineDir: Basics.AbsolutePath, modulesToDiff: Set<String>) async throws {
```

Comment:
suggestion: Consider created automated tests that validate this function in isolation with happy path, and also fault injection, and more.